pkg/kp/rcstore/consul_store.go

```diff
@@ -138,31 +138,24 @@ func (s *consulStore) List() ([]fields.RC, error) {
 func (s *consulStore) WatchNew(quit <-chan struct{}) (<-chan []fields.RC, <-chan error) {
 	outCh := make(chan []fields.RC)
 	errCh := make(chan error)
+	inCh := make(chan api.KVPairs)
+
+	go consulutil.WatchPrefix(kp.RC_TREE, s.kv, inCh, quit, errCh)
```

Comment:
good point - how about blocking on the goroutine exit until the kvp channel is closed?